butterfree/load/writers/delta_writer.py

```diff
@@ -57,6 +61,15 @@ def merge(
             source.<column_name> and target.<column_name>
         """
         try:
+
+            df_to_merge = source_df
+            if deduplicate:
+                if feature_set is None:
+                    raise ValueError(
+                        "feature_set must be provided when deduplicate=True"
+                    )
+                df_to_merge = feature_set._filter_duplicated_rows(source_df)
```

Comment:
I'm not suuuper comfortable with the idea having this deduplication happening at this level for two reasons:

1. deduplication is essentially another transformation, so maybe we should left this responsibility on the feature set level?
2. `_filter_duplicated_rows` is a feature set private method that we should try to avoid using outside of the feature set context to avoid unnecessary coupling.

Should we just assume that whenever this delta writer is going to be used with its MERGE operations that the dedup is handled on the feature set level?